### PR TITLE
Docker updates

### PR DESCRIPTION
**Pull request by yolile**
To be merged after:
- https://github.com/open-contracting/data-registry/pull/170
- https://github.com/open-contracting/pelican-backend/pull/44
- https://github.com/open-contracting/pelican-frontend/pull/68
- https://github.com/open-contracting/kingfisher-process/pull/344

### CODE REVIEW

File: salt/docker_apps/files/pelican_backend.yaml [hunk 1]
```diff
@@ -2,11 +2,6 @@ version: "2.2"
 
 x-shared: &shared
   image: "ghcr.io/open-contracting/pelican-backend:latest"
-  network_mode: "host"
```
Comment (jpmckinney) [inline salt/docker_apps/files/pelican_backend.yaml:5]: Pelican backend needs to communicate with PostgreSQL and RabbitMQ on the host, and it needs to make remote connections. Can it still do those without `network_mode: "host"`?

**Review by jpmckinney — COMMENTED** [review]
Also need to update https://github.com/open-contracting/deploy/blob/main/salt/docker_apps/init.sls#L19-L21

File: salt/docker_apps/files/kingfisher_process.yaml [hunk 2]
```diff
@@ -3,8 +3,6 @@ version: "2.2"
 x-shared: &shared
   image: "ghcr.io/open-contracting/kingfisher-process:latest"
   network_mode: "host"
-  env_file:
-    - env
   volumes:
     - /data:/data
```
Comment (jpmckinney) [inline salt/docker_apps/files/kingfisher_process.yaml:7]: Should this also be /data/storage? (I haven't checked if it reads outside there.)

Comment (yolile) [reply salt/docker_apps/files/kingfisher_process.yaml:7]: I didn't check for the kingfisher-process Dockerfile yet. From a quick look, I see the /data directory is created in the Dockerfile. From the deploy repo I see https://github.com/open-contracting/deploy/blob/9298e0690da0190dd62221a84d6b2ff0d1c14689/pillar/registry.sls#L54 but we are not setting the `KINGFISHER_API_LOCAL_DIRECTORY` as in https://github.com/open-contracting/deploy/blob/9298e0690da0190dd62221a84d6b2ff0d1c14689/pillar/kingfisher.sls#L78, so I'm not sure how kingfisher-collect and kingfisher-process are connected in the registry server and therefore, for what the /data directory is used in kingfisher-process.

File: salt/docker_apps/files/pelican_backend.yaml (hunk 1 above)
Comment (yolile) [reply salt/docker_apps/files/pelican_backend.yaml:5]: Oh, I forgot about that. We could do the same as https://github.com/open-contracting/deploy/pull/316/commits/4cf8abb7b450f021eb2ab746e9e002ac9180c111 to still be able to communicate with the host's Postgres and RabbitMQ services, or we can leave this the network_mose as "host".

File: salt/docker_apps/files/kingfisher_process.yaml (hunk 2 above)
Comment (jpmckinney) [reply salt/docker_apps/files/kingfisher_process.yaml:7]: `KINGFISHER_API_LOCAL_DIRECTORY` is only required for the old Process integration. It is not used by the new Process integration. In both cases, however, Process reads the JSON files written by Collect.

File: salt/docker_apps/files/pelican_backend.yaml (hunk 1 above)
Comment (jpmckinney) [reply salt/docker_apps/files/pelican_backend.yaml:5]: We should have the same approach to `network_mode` across all files. We can either change it now to use the approach in 4cf8abb, or open a separate issue.

File: salt/docker_apps/files/kingfisher_process.yaml (hunk 2 above)
Comment (yolile) [reply salt/docker_apps/files/kingfisher_process.yaml:7]: So Kingfisher process will need to have access to /data/storage/kingfisher-collect? If so we can update KF to create that same directory in its Dockerfile and then update the volume here or leave it as it is now

Comment (jpmckinney) [reply salt/docker_apps/files/kingfisher_process.yaml:7]: Yes, it needs that access. We might as well update its Dockerfile to make it follow the common template: https://ocp-software-handbook.readthedocs.io/en/latest/docker/index.html#templates

Comment (yolile) [reply salt/docker_apps/files/kingfisher_process.yaml:7]: Great to see that you made a template now!

Comment (yolile) [reply salt/docker_apps/files/kingfisher_process.yaml:7]: done in https://github.com/open-contracting/kingfisher-process/pull/344

File: salt/docker_apps/files/pelican_backend.yaml (hunk 1 above)
Comment (yolile) [reply salt/docker_apps/files/pelican_backend.yaml:5]: I will leave the network_mode as it is here, and then we can follow up on the existing issue https://github.com/open-contracting/deploy/issues/307

**Comment by yolile** [comment]
I've updated this and the related PRs to use the defaults directories instead of defining them twice: once in the env list and then again in the volumes declaration, as we agreed in our last call

File: salt/docker_apps/files/kingfisher_process.yaml [hunk 3]
```diff
@@ -3,10 +3,8 @@ version: "2.2"
 x-shared: &shared
   image: "ghcr.io/open-contracting/kingfisher-process:latest"
   network_mode: "host"
-  env_file:
-    - env
   volumes:
-    - /data:/data
+    - {{ pillar.kingfisher_collect.files_store }}:/data
```
Comment (jpmckinney) [inline salt/docker_apps/files/kingfisher_process.yaml:7]: Why not `pillar.kingfisher_collect.env.FILES_STORE` to avoid creating another Pillar key?

Comment (yolile) [reply salt/docker_apps/files/kingfisher_process.yaml]: True, changed it now

**Review by jpmckinney — APPROVED** [review]

**Pull request merged**